src_python/habitat_sim/simulator.py

```diff
@@ -277,48 +338,312 @@ def __set_from_config(self, config: Configuration) -> None:
+                    sensor_spec.resolution[0],
+                    sensor_spec.resolution[1],
+                    dtype=torch.float32,
+                    device=device,
+                )
+            else:
+                self.__obs_buffers_[sensor_spec.uuid] = torch.empty(
+                    sensor_spec.resolution[0],
+                    sensor_spec.resolution[1],
+                    sensor_spec.channels,  # for R, G, B, A
+                    dtype=torch.uint8,
+                    device=device,
+                )
+        else:
+            # create empty ndarry to store sensor observations
+            # create empty MutableImageView2D to store results of what is in render target
+            view_size = sensor.framebuffer_size
+
+            if sensor_spec.sensor_type == SensorType.SEMANTIC:
+                self.__obs_buffers_[sensor_spec.uuid] = np.empty(
+                    (sensor_spec.resolution[0], sensor_spec.resolution[1]),
+                    dtype=np.uint32,
+                )
+
+                obs_buffer = self.__obs_buffers_.get(sensor_spec.uuid)
+                self.__image_views_[sensor_spec.uuid] = mn.MutableImageView2D(
+                    mn.PixelFormat.R32UI, view_size, obs_buffer
+                )
+            elif sensor_spec.sensor_type == SensorType.DEPTH:
+                self.__obs_buffers_[sensor_spec.uuid] = np.empty(
+                    (sensor_spec.resolution[0], sensor_spec.resolution[1]),
+                    dtype=np.float32,
+                )
+
+                obs_buffer = self.__obs_buffers_.get(sensor_spec.uuid)
+                self.__image_views_[sensor_spec.uuid] = mn.MutableImageView2D(
+                    mn.PixelFormat.R32F,
+                    view_size,
+                    obs_buffer,
+                )
+            else:  # sensor_spec.sensor_type == SensorType.COLOR:
+                self.__obs_buffers_[sensor_spec.uuid] = np.empty(
+                    (
+                        sensor_spec.resolution[0],
+                        sensor_spec.resolution[1],
+                        sensor_spec.channels,  # for R, G, B, A
+                    ),
+                    dtype=np.uint8,
+                )
+
+                obs_buffer = self.__obs_buffers_.get(sensor_spec.uuid)
+                self.__image_views_[sensor_spec.uuid] = mn.MutableImageView2D(
+                    mn.PixelFormat.RGBA8_UNORM,
+                    view_size,
+                    obs_buffer.reshape(sensor_spec.resolution[0], -1),  # type: ignore[union-attr]
+                )
+
+    def get_sensor(self, sensor_uuid, agent_id: Optional[int] = None) -> Sensor:
+        if agent_id is not None:
+            return self.__sensors[agent_id].get(sensor_uuid)
+        else:
+            # TODO: use this eventually, so that sensors are independent of agents
+            return self.__sensors_.get(sensor_uuid)
+
+    def get_subtree_sensors(self, scene_node: SceneNode) -> Dict[str, Sensor]:
```

Comment:
If scene_node can be none, the arg should be Optional[SceneNode], no?

Reply:
Not sure what you mean here. The code is asserting that `scene_node` is not none.